network-ntt/src/server.rs

```diff
@@ -0,0 +1,313 @@
+//! Network transport node, implementes an interface to the
+//! chain node, based on the network-transport protocol (NT later).
+//! NT is a simplified protocol that mimic CCI, a protocol
+//! that support differrent quality networks and provides a high
+//! level interface for the mutiplexing uni-directional connections
+//! between the nodes.
+//!
+//! For better understanding of the network transport concept
+//! you may check [document]().
+//!
+//! Node may act both as a server and a client. Server listens
+//! on the specified port and client connects to the remote service.
+//! Once node node establised a connection to the other one, the other
+//! once can build an opposite connection and nodes can talk to
+//! each other.
+//!
+//! ```text
+//!                                     +---- server: sock1
+//!                                     |      |
+//!                                     |      |-- connection1
+//!                                     |      |-- connection2
+//!  +--------------+     +----------+  |
+//!  | network-core | ----| NT state |--+---- server: sock2
+//!  +--------------+     +----------+  |      |
+//!                                     |      |-- connection3
+//!                                     |
+//!                                     +---- client: remote-sock1
+//! ```
+//!
+//! As NT state is shared all the services could provide a uniform
+//! access to the chain state. And one could build very flexible
+//! topology.
+//use crate::reply_handle::ReplyHandle;
+//use chain_core::property::{Deserialize, Serialize};
+use futures::prelude::*;
+use futures::{future, stream::{self,Stream}, sync::mpsc};
+use network_core::server;
+use network_core::server::block::BlockService;
+use network_core::server::transaction::TransactionService;
+use std::net::SocketAddr;
+use tokio::net::{TcpListener, TcpStream};
+use tokio_protocol::{protocol, Connection, Inbound, Message};
+
+// These types are loaded from cardano package,
+// because tokio-protocol is not generalized yet.
+// This should be fixed and then these types should go.
+use cardano::{
+    block::{HeaderHash},
+    tx::{TxAux},
+};
+
+/// Network transport server configuration.
+pub struct Config {
```

Comment:
I did not bring a `Config` into the `network-grpc` library. Individual calls to `server::listen` and a future `server::connect` can be used to instantiate connections. I think we could live it up to the application how to manage collections of peers, unless we have a generally perfect idea for everyone.